pkg/operator/baremetal_config.go

```diff
@@ -65,3 +77,89 @@ func getBaremetalProvisioningConfig(dc dynamic.Interface, configName string) (Ba
 		ProvisioningDHCPRange:    provisioningDHCPRange,
 	}, nil
 }
+
+func getProvisioningIPCIDR(baremetalConfig BaremetalProvisioningConfig) string {
+	if baremetalConfig.ProvisioningIp != "" {
+		cidr := strings.Split(baremetalConfig.ProvisioningNetworkCIDR, "/")
+		return baremetalConfig.ProvisioningIp + "/" + cidr[1]
+	} else {
+		return unknownProvisioningConfig
+	}
+}
+
+func getDeployKernelUrl(baremetalConfig BaremetalProvisioningConfig) string {
+	if baremetalConfig.ProvisioningIp != "" {
+		return baremetalConfig.ProvisioningIp + ":" + baremetalHttpPort + baremetalKernelUrlSubPath
+	} else {
+		return unknownProvisioningConfig
+	}
+}
+
+func getDeployRamdiskUrl(baremetalConfig BaremetalProvisioningConfig) string {
+	if baremetalConfig.ProvisioningIp != "" {
+		return baremetalConfig.ProvisioningIp + ":" + baremetalHttpPort + baremetalRamdiskUrlSubPath
+	} else {
+		return unknownProvisioningConfig
+	}
+}
+
+func getIronicEndpoint(baremetalConfig BaremetalProvisioningConfig) string {
+	if baremetalConfig.ProvisioningIp != "" {
+		return baremetalConfig.ProvisioningIp + ":" + baremetalIronicPort + baremetalIronicEndpointSubpath
+	} else {
+		return unknownProvisioningConfig
+	}
+}
+
+func getIronicInspectorEndpoint(baremetalConfig BaremetalProvisioningConfig) string {
+	if baremetalConfig.ProvisioningIp != "" {
+		return baremetalConfig.ProvisioningIp + ":" + baremetalIronicInspectorPort + baremetalIronicEndpointSubpath
+	} else {
+		return unknownProvisioningConfig
+	}
+}
+
+func getProvisioningDHCPRange(baremetalConfig BaremetalProvisioningConfig) string {
+	if baremetalConfig.ProvisioningIp == "" {
+		return unknownProvisioningConfig
+	}
+	if !baremetalConfig.ProvisioningDHCPExternal && baremetalConfig.ProvisioningDHCPRange == "" {
+		cidr := strings.Split(baremetalConfig.ProvisioningNetworkCIDR, "/")
+		partialProvisioningIp := strings.Split(cidr[0], ".")
+		provisioningDHCPRangeStart := partialProvisioningIp[0] + "." + partialProvisioningIp[1] + "." + partialProvisioningIp[2] + "." + "10"
+		provisioningDHCPRangeEnd := partialProvisioningIp[0] + "." + partialProvisioningIp[1] + "." + partialProvisioningIp[2] + "." + "100"
+		baremetalConfig.ProvisioningDHCPRange = provisioningDHCPRangeStart + ", " + provisioningDHCPRangeEnd
```

Comment:
ProvisioningDHCPRange will always be set in the CR if DHCP isn't external, it's done by the installer's defaults setting code: https://github.com/openshift/installer/blob/master/pkg/types/baremetal/defaults/platform.go#L44-L46

MAO shouldn't have any logic to handle this. If it's empty, it's empty (but it shouldn't ever be).  

Also, the way you're constructing it here doesn't work with IPv6. You can see how the installer does it using cidr.Host.

Reply:
Removed code to generate default DHCP range since the installer is taking care of this step.